source/csharp/Camera/Basic/CaptureViaZivid/CaptureViaZivid.cs

```diff
@@ -97,29 +109,43 @@ private static HalconDotNet.HObjectModel3D ZividToHalconPointCloud(Zivid.NET.Poi
         {
             for (uint j = 0; j < width; j++)
             {
-                double x = pointsXYZ[i, j, 0];
+                float x = pointsXYZ[i, j, 0];
+                float normal = normalsXYZ[i, j, 0];
 
                 if (!Double.IsNaN(x))
                 {
-                    tupleX[validPointIndex] = pointsXYZ[i, j, 0];
-                    tupleY[validPointIndex] = pointsXYZ[i, j, 1];
-                    tupleZ[validPointIndex] = pointsXYZ[i, j, 2];
+                    tuplePointsX[validPointIndex] = pointsXYZ[i, j, 0];
+                    tuplePointsY[validPointIndex] = pointsXYZ[i, j, 1];
+                    tuplePointsZ[validPointIndex] = pointsXYZ[i, j, 2];
                     tupleR[validPointIndex] = colorsRGBA[i, j, 0];
                     tupleG[validPointIndex] = colorsRGBA[i, j, 1];
                     tupleB[validPointIndex] = colorsRGBA[i, j, 2];
                     tupleXYZMapping[2 + validPointIndex] = i;
                     tupleXYZMapping[2 + numberOfValidPoints + validPointIndex] = j;
+
+                    if (!Double.IsNaN(normal))
+                    {
+                        tupleNormalsX[validPointIndex] = normalsXYZ[i, j, 0];
+                        tupleNormalsY[validPointIndex] = normalsXYZ[i, j, 1];
+                        tupleNormalsZ[validPointIndex] = normalsXYZ[i, j, 2];
+                    }
+
                     validPointIndex++;
                 }
             }
         }
 
         Console.WriteLine("Constructing ObjectModel3D based on XYZ data");
-        var objectModel3D = new HalconDotNet.HObjectModel3D(tupleX, tupleY, tupleZ);
+        var objectModel3D = new HalconDotNet.HObjectModel3D(tuplePointsX, tuplePointsY, tuplePointsZ);
 
         Console.WriteLine("Mapping ObjectModel3D data");
         HalconDotNet.HOperatorSet.SetObjectModel3dAttribMod(objectModel3D, "xyz_mapping", "object", tupleXYZMapping);
 
+        Console.WriteLine("Adding normals to ObjectModel3D");
+        var attribNames = new HalconDotNet.HTuple("point_normal_x", "point_normal_y", "point_normal_z");
+        var attribValues = new HalconDotNet.HTuple(tupleNormalsX, tupleNormalsY, tupleNormalsZ);
```

Comment:
maybe add "normal" or "normals" to "attribNames" and "attribValues" because line 

`HalconDotNet.HOperatorSet.SetObjectModel3dAttribMod(objectModel3D, attribNames, "points", attribValues);`

will be more clear if it is like:

`HalconDotNet.HOperatorSet.SetObjectModel3dAttribMod(objectModel3D, normalsAttribNames, "points", normalAttribValues);`